From b497198739be9eab261b0a3bdf56aa8107c5771b Mon Sep 17 00:00:00 2001
From: Bruno Martins <bgcngm@gmail.com>
Date: Fri, 16 Nov 2018 11:12:37 +0000
Subject: [PATCH 3/3] sepolicy: Grant mediaextractor access to files over all
 sdcard fs types

 * Account for all possible sdcard-related filesystem types and not only
   sdcardfs:

   public/file.te:108:type fuse, sdcard_type, fs_type, mlstrustedobject;
   public/file.te:109:type sdcardfs, sdcard_type, fs_type, mlstrustedobject;
   public/file.te:110:type sdcard_posix, sdcard_type, fs_type, mlstrustedobject;
   public/file.te:111:type vfat, sdcard_type, fs_type, mlstrustedobject;
   public/file.te:112:type exfat, sdcard_type, fs_type, mlstrustedobject;

Change-Id: Ic508397bf4ca66a002ada33ac3f600c17b8a1a10
---
 public/mediaextractor.te | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/public/mediaextractor.te b/public/mediaextractor.te
index b0554626..4c40850f 100644
--- a/public/mediaextractor.te
+++ b/public/mediaextractor.te
@@ -23,7 +23,7 @@ allow mediaextractor proc_meminfo:file r_file_perms;
 crash_dump_fallback(mediaextractor)
 
 # allow mediaextractor read permissions for file sources
-allow mediaextractor sdcardfs:file { getattr read };
+allow mediaextractor sdcard_type:file { getattr read };
 allow mediaextractor media_rw_data_file:file { getattr read };
 allow mediaextractor app_data_file:file { getattr read };
 
-- 
2.17.1

